src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariants.java

```diff
@@ -0,0 +1,1244 @@
+     * the genotype level.
+     */
+    @Argument(fullName="minFractionFilteredGenotypes", optional=true,
+                    doc="Maximum fraction of samples filtered at the genotype level")
+    public double minFractionFilteredGenotypes = MIN_FRACTION_FILTERED_GENOTYPES_DEFAULT_VALUE;
+
+    /**
+     * If this argument is provided, set filtered genotypes to no-call (./.).
+     */
+    @Argument(fullName="setFilteredGtToNocall", optional=true, doc="Set filtered genotypes to no-call")
+    public boolean setFilteredGenotypesToNocall = false;
+
+    @Argument(fullName=StandardArgumentDefinitions.LENIENT_LONG_NAME,
+                            shortName = StandardArgumentDefinitions.LENIENT_SHORT_NAME,
+                            doc = "Lenient processing of VCF files", common = true, optional = true)
+    public boolean lenientVCFProcesssing;
+
+    //@ArgumentCollection
+    //private IntervalArgumentCollection intervalList = new OptionalIntervalArgumentCollection();
+
+    @HiddenOption
+    @Argument(fullName="ALLOW_NONOVERLAPPING_COMMAND_LINE_SAMPLES", optional=true,
+                    doc="Allow samples other than those in the VCF to be specified on the command line. These samples will be ignored.")
+    public boolean allowNonOverlappingCommandLineSamples = false;
+
+    @HiddenOption
+    @Argument(fullName="SUPPRESS_REFERENCE_NAME", shortName="sr", optional=true,
+            doc="Suppress reference file name in output for test result differencing")
+    public boolean suppressReferenceName = false;
+
+    private VariantContextWriter vcfWriter = null;
+
+    private enum NumberAlleleRestriction {
+        ALL,
+        BIALLELIC,
+        MULTIALLELIC
+    }
+
+    private TreeSet<String> samples = new TreeSet<>();
+    private boolean noSamplesSpecified = false;
+
+    private Set<VariantContext.Type> selectedTypes = new HashSet<>();
+    private ArrayList<String> selectNames = new ArrayList<>();
+    private List<VariantContextUtils.JexlVCMatchExp> jexls = null;
+
+    private boolean discordanceOnly = false;
+    private boolean concordanceOnly = false;
+
+    private MendelianViolation mv = null;
+    private SampleDB sampleDB = null;
+
+    /* variables used by the SELECT RANDOM modules */
+    private boolean selectRandomFraction = false;
+
+    // Random number generator for the genotypes to remove
+    private Random randomGenotypes = new Random();
+
+    private Set<String> IDsToKeep = null;
+    private Set<String> IDsToRemove = null;
+
+    private final List<Allele> diploidNoCallAlleles = Arrays.asList(Allele.NO_CALL, Allele.NO_CALL);
+
+    /**
+     * Set up the VCF writer, the sample expressions and regexs, filters inputs, and the JEXL matcher
+     *
+     */
+    @Override
+    public void onTraversalStart() {
+        final FeatureInput<? extends Feature> featureInput = getDrivingVariantsFeatureInput();
+
+        final Map<String, VCFHeader> vcfRods = new HashMap<>();
```

Comment:
Name this something like `vcfHeaders` -- "Rods" is an obsolete and confusing term